### PR TITLE
Feat/41 migrate to ts template

### PR DESCRIPTION
**Pull request by horaklukas**
Contains
---------
* Merge most recent template from `cra-template` into `cra-tempalate-typescript`
* Reset `cra-template` into the current state of [default facebook's `cra-template`](https://github.com/facebook/create-react-app/tree/main/packages/cra-template)
* Fix typing of source components (mostly of HOCs) and other cor utils (enhancers, store, etc.)
* Adjust tsconfig according to what worked in other projects
*  Add `static.d.ts` with static assets declarations

Not includes - will be in next MR
--------------------------------
* Upgrade of `@ackee/eslint-config` to the latest version
* Configure of eslint to work with TS

Notes
------
For new or refactorted imports the pattern of import from dependencies files wasn't followed as it's anyway intended to remove in the future.

Relates #41

### CODE REVIEW

File: packages/cra-template-typescript/CHANGELOG.md [hunk 1]
```diff
@@ -18,10 +18,12 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
...
 - Min required node version is 10.
 
+## @ackee/cra-template@1.0.1 - 2020-04-30
+
 ## Fixed
 
 - Use `BUILD_ENV` instead of `NODE_ENV` for retrieving correct env. config.
 
-## @ackee/cra-template@0.0.1 - 2020-01-17
+## @ackee/cra-template@1.0.0 - 2020-01-09
```
Comment (cermakjiri) [inline packages/cra-template-typescript/CHANGELOG.md]: Is this typo?

File: packages/cra-template-typescript/template/.env [hunk 2]
```diff
@@ -12,4 +12,3 @@ CHOKIDAR_USEPOLLING=false
 # This is useful for example when you're developing an NPM package. 
 # Should be disabled otherwise to prevent slower rebuilds. 
 REACT_WATCH_NODE_MODULES=false
```
Comment (cermakjiri) [inline packages/cra-template-typescript/template/.env:14]: Please disable FAST_REFRESH, it's experimental and doesn't work for now:
```
FAST_REFRESH=false
```

File: packages/cra-template-typescript/template/gitignore [hunk 3]
```diff
@@ -17,15 +17,7 @@
...
 
 npm-debug.log*
 yarn-debug.log*
 yarn-error.log*
-
-.yalc
-yalc.lock
-
-# localization files are generated, it's not source thus don't version them
-src/translations/*.json
```
Comment (cermakjiri) [inline packages/cra-template-typescript/template/gitignore:31]: Let's not removed these values, I think we want to still ignore them

File: packages/cra-template-typescript/CHANGELOG.md (hunk 1 above)
Comment (horaklukas) [reply packages/cra-template-typescript/CHANGELOG.md]: I copy pasted `cra-template` to `cra-template-typescript` so these code unrelated changes are caused by that

File: packages/cra-template-typescript/template/.env (hunk 2 above)
Comment (horaklukas) [reply packages/cra-template-typescript/template/.env:14]: same as above ☝️

File: packages/cra-template-typescript/template/gitignore (hunk 3 above)
Comment (horaklukas) [reply packages/cra-template-typescript/template/gitignore:31]: same as above ☝️

File: packages/cra-template-typescript/CHANGELOG.md (hunk 1 above)
Comment (cermakjiri) [reply packages/cra-template-typescript/CHANGELOG.md]: Oh, right 👍

File: packages/cra-template-typescript/template/.env (hunk 2 above)
Comment (cermakjiri) [reply packages/cra-template-typescript/template/.env:14]: I remembered that it's required to have it there (applies for `@ackee/react-scripts@2.x`). I know it's not the direct subject of this PR. Could you please add it there anyway?

File: packages/cra-template-typescript/template/gitignore (hunk 3 above)
Comment (cermakjiri) [reply packages/cra-template-typescript/template/gitignore:31]: Regardless, we still want to have it there.

File: packages/cra-template-typescript/template/.env (hunk 2 above)
Comment (horaklukas) [reply packages/cra-template-typescript/template/.env:14]: I added it

File: packages/cra-template-typescript/template/gitignore (hunk 3 above)
Comment (horaklukas) [reply packages/cra-template-typescript/template/gitignore:31]: I reverted it

**Review by cermakjiri — APPROVED** [review]

**Pull request merged**